.github/workflows/go.yml

```diff
@@ -0,0 +1,34 @@
+name: Build and test Go
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: "*"
+env:
+    GO111MODULE: on
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.12.x
+      - name: Cache Go modules
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: build-${{ hashFiles('**/go.sum') }}
+      - name: Build
+        run: go build -v
+        working-directory: ./chaincode
+      - name: Test
+        run: go test -v ./...
+        working-directory: ./chaincode
+      - name: Lint
```

Comment:
You can run the lint first in this testing pipeline, that way we fail early without running the potentially costly operation of building the go package (build takes `13s` vs `2s` for the lint) if the code does not comply with the lint rules.